lib/util/uint8array_utils.js

```diff
@@ -131,11 +131,49 @@ shaka.util.Uint8ArrayUtils.equal = function(arr1, arr2) {
 
 
 /**
- * Convert a Uint8Array to a string which can be used as a key in a dictionary.
+ * Generates a random unique number to use as a key for a given Unit8Array
+ * or returns the key for a a given Unit8Array
  * @param {!Uint8Array} arr
  * @return {string}
  */
-shaka.util.Uint8ArrayUtils.key = function(arr) {
-  return Array.prototype.join.apply(arr);
-};
-
+shaka.util.Uint8ArrayUtils.key = (function() {
+
+  var cache = {};
```

Comment:
We should find another way to generate these keys.  This cache is essentially just a big memory leak.  Consider key rotation in live streams.  There could be a new unique init datas every X seconds during a live event for an indefinite period of time.